File: bone-business/tpa-saas/bone-lowcode/bone-lowcode-infra/src/main/java/com/bone/lowcode/infra/domain/valueobject/EventTypeEnum.java
package com.bone.lowcode.infra.domain.valueobject;

import lombok.Getter;

//事件类型枚举
@Getter
public enum EventTypeEnum {
    PROCESS((byte) 0, "流程事件"),
    BUSINESS((byte)1, "业务事件");
    private final byte code;
    private final String desc;

    EventTypeEnum(byte code, String desc) {
        this.code = code;
        this.desc = desc;
    }
}

File: bone-business/tpa-sass-vue3/src/enums/event/EventTypeEnum.ts
export enum EventTypeEnum {
  process = 0,
  business = 1,
}

// 事件类型标签
export const EventTypeLabels = {
  [EventTypeEnum.process]: "流程事件",
  [EventTypeEnum.business]: "业务事件",
};

// 事件类型选项
export const EventTypeOptions = [
  {
    label: EventTypeLabels[EventTypeEnum.process],
    value: EventTypeEnum.process,
  },
  {
    label: EventTypeLabels[EventTypeEnum.business],
    value: EventTypeEnum.business,
  },
];

// 获取事件类型标签
export const getEventTypeLabel = (value: number | EventTypeEnum) => {
  return EventTypeLabels[value as EventTypeEnum];
};
